falcon/http_error.py

```diff
@@ -194,9 +204,10 @@ def to_json(self, handler=None):
         obj = self.to_dict(OrderedDict)
         if handler is None:
             handler = _DEFAULT_JSON_HANDLER
```

Comment:
we can probably type `_DEFAULT_JSON_HANDLER` like this so there is no need of the assert
```
_DEFAULT_JSON_HANDLER: Serializer
```

Reply:
Done. Have a look, because I don't know if this is what you meant.